src/overlays/actors/ovl_En_Wood02/z_en_wood02.c

```diff
@@ -28,29 +50,466 @@ const ActorInit En_Wood02_InitVars = {
+        }
+        this->unk_151 = 1;
+    } else {
+        this->actor.child = NULL;
+        this->unk_151 = 0;
+    }
+
+    switch (this->actor.params) {
+        case WOOD_BUSH_GREEN_LARGE_SPAWNER:
+        case WOOD_BUSH_BLACK_LARGE_SPAWNER:
+            spawnType = 1;
+
+        case WOOD_BUSH_GREEN_LARGE_SPAWNED:
+        case WOOD_BUSH_BLACK_LARGE_SPAWNED:
+            spawnType++;
+
+        case WOOD_TREE_CONICAL_LARGE:
+        case WOOD_BUSH_GREEN_LARGE:
+        case WOOD_BUSH_BLACK_LARGE:
+        case WOOD_1A:
+            actorScale = 1.5f;
+            this->actor.uncullZoneForward = 4000.0f;
+            this->actor.uncullZoneScale = 2000.0f;
+            this->actor.uncullZoneDownward = 2400.0f;
+            break;
+
+        case WOOD_TREE_CONICAL_SPAWNER:
+        case WOOD_TREE_OVAL_YELLOW_SPAWNER:
+        case WOOD_TREE_OVAL_GREEN_SPAWNER:
+        case WOOD_BUSH_GREEN_SMALL_SPAWNER:
+        case WOOD_BUSH_BLACK_SMALL_SPAWNER:
+            spawnType = 1;
+
+        case WOOD_TREE_CONICAL_SPAWNED:
+        case WOOD_TREE_OVAL_YELLOW_SPAWNED:
+        case WOOD_TREE_OVAL_GREEN_SPAWNED:
+        case WOOD_BUSH_GREEN_SMALL_SPAWNED:
+        case WOOD_BUSH_BLACK_SMALL_SPAWNED:
+            spawnType++;
+
+        case WOOD_TREE_CONICAL_MEDIUM:
+        case WOOD_TREE_OVAL_GREEN:
+        case WOOD_TREE_KAKARIKO_ADULT:
+        case WOOD_BUSH_GREEN_SMALL:
+        case WOOD_BUSH_BLACK_SMALL:
+            this->actor.uncullZoneForward = 4000.0f;
+            this->actor.uncullZoneScale = 800.0f;
+            this->actor.uncullZoneDownward = 1800.0f;
+            break;
+
+        case WOOD_TREE_CONICAL_SMALL:
+            actorScale = 0.6f;
+            this->actor.uncullZoneForward = 4000.0f;
+            this->actor.uncullZoneScale = 400.0f;
+            this->actor.uncullZoneDownward = 1000.0f;
+            break;
+
+        case WOOD_LEAF_GREEN:
+        case WOOD_LEAF_YELLOW:
+            this->unk_14A[0] = 75;
+            actorScale = 0.02f;
+            this->actor.velocity.x = randPlusMinusPoint5Scaled(6.0f);
+            this->actor.velocity.z = randPlusMinusPoint5Scaled(6.0f);
+            this->actor.velocity.y = (Rand_ZeroOne() * 1.25f) + -3.1f;
+            break;
+    }
+
+    if (this->actor.params < WOOD_TREE_OVAL_GREEN) {
+        THIS->drawType = WOOD_DRAW_TREE_CONICAL;
+    } else if (this->actor.params < WOOD_TREE_KAKARIKO_ADULT) {
+        THIS->drawType = WOOD_DRAW_TREE_OVAL;
+    } else if (this->actor.params < WOOD_BUSH_GREEN_SMALL) {
+        THIS->drawType = WOOD_DRAW_TREE_KAKARIKO_ADULT;
+    } else if (this->actor.params < WOOD_BUSH_BLACK_SMALL) {
+        THIS->drawType = WOOD_DRAW_BUSH_GREEN;
+    } else if (this->actor.params < WOOD_LEAF_YELLOW) {
+        THIS->drawType = WOOD_DRAW_4;
+    } else if (this->actor.params == WOOD_1A) {
+        THIS->drawType = WOOD_DRAW_TREE_CONICAL;
+    } else {
+        THIS->drawType = WOOD_DRAW_LEAF_YELLOW;
+    }
```

Comment:
Is there no other way?

Reply:
Whatr do you mean?

Reply:
Than using `THIS`, which is strongly discouraged.

Reply:
https://decomp.me/scratch/edcey
Seems to match?

Reply:
This doesn't seem to have been addressed.